addons/website/static/src/js/utils.js

```diff
@@ -239,6 +239,27 @@ function websiteDomain(self) {
     return ['|', ['website_id', '=', false], ['website_id', '=', websiteID]];
 }
 
+/**
+ * Checks if the 2 given URLs originates from the same domain.
+ * It will consider naked URL and `www` URL originating from the same domain.
+ * It will consider `https` URL `http` URL originating from the same domain.
+ *
+ * @param {string} url1
+ * @param {string} url2
+ * @returns {Boolean}
+ */
+function isHTTPSorNakedDomainRedirection(url1, url2) {
+    try {
+        url1 = new URL(url1).host;
+        url2 = new URL(url2).host;
+    } catch {
+        // Incorrect URL, `false` URL..
+        return false;
+    }
+    return url1 === url2 ||
+           url1.replace(/^www./, '') === url2.replace(/^www./, '');
```

Comment:
`/^www\./` (need to escape the `.` as discussed in discord :wink:)

Reply:
Done, see https://github.com/odoo/odoo/compare/e1bbfa27bf121e615e13972faba033f0e90d2d21..f2a2866679331d3fa0529c6e5466c432662bec79